tracer/src/Datadog.Trace/Ci/TestSuite.cs

```diff
@@ -0,0 +1,176 @@
+
+        if (startDate is null)
+        {
+            // If a module doesn't have a fixed start time we reset it before running code
+            span.ResetStartTime();
+        }
+    }
+
+    /// <summary>
+    /// Gets the current TestSuite
+    /// </summary>
+    public static TestSuite? Current
+    {
+        get => CurrentSuite.Value;
+        internal set => CurrentSuite.Value = value;
+    }
+
+    /// <summary>
+    /// Gets the test suite name
+    /// </summary>
+    public string Name { get; }
+
+    /// <summary>
+    /// Gets the test suite start date
+    /// </summary>
+    public DateTimeOffset StartDate => _span.StartTime;
+
+    /// <summary>
+    /// Gets the test module for this suite
+    /// </summary>
+    public TestModule Module { get; }
+
+    internal TestSuiteSpanTags Tags => (TestSuiteSpanTags)_span.Tags;
+
+    /// <summary>
+    /// Sets a string tag into the test
+    /// </summary>
+    /// <param name="key">Key of the tag</param>
+    /// <param name="value">Value of the tag</param>
+    public void SetTag(string key, string? value)
+    {
+        _span.SetTag(key, value);
+    }
+
+    /// <summary>
+    /// Sets a number tag into the test
+    /// </summary>
+    /// <param name="key">Key of the tag</param>
+    /// <param name="value">Value of the tag</param>
+    public void SetTag(string key, double? value)
+    {
+        _span.SetMetric(key, value);
+    }
+
+    /// <summary>
+    /// Set Error Info
+    /// </summary>
+    /// <param name="type">Error type</param>
+    /// <param name="message">Error message</param>
+    /// <param name="callStack">Error callstack</param>
+    public void SetErrorInfo(string type, string message, string? callStack = null)
+    {
+        var span = _span;
+        span.Error = true;
+        span.SetTag(Trace.Tags.ErrorType, type);
+        span.SetTag(Trace.Tags.ErrorMsg, message);
+        if (callStack is not null)
+        {
+            span.SetTag(Trace.Tags.ErrorStack, callStack);
+        }
+    }
+
+    /// <summary>
+    /// Set Error Info from Exception
+    /// </summary>
+    /// <param name="exception">Exception instance</param>
+    public void SetErrorInfo(Exception exception)
+    {
+        _span.SetException(exception);
+    }
+
+    /// <summary>
+    /// Close test suite
+    /// </summary>
+    /// <param name="duration">Duration of the test suite</param>
+    public void Close(TimeSpan? duration = null)
+    {
+        if (Interlocked.Exchange(ref _finished, 1) == 1)
+        {
+            return;
+        }
+
+        var span = _span;
+
+        // Calculate duration beforehand
+        duration ??= span.Context.TraceContext.ElapsedSince(span.StartTime);
+
+        // Update status
+        if (Tags.Status is { } status)
+        {
+            if (status == TestTags.StatusFail)
+            {
+                Module.Tags.Status = TestTags.StatusFail;
+            }
```

Comment:
Should we handle skipped here too?

Reply:
No, we want to fail a suite or a module if a test failed, but a test suite should pass if we have test being skipped.